src/OlStyler.js

```diff
@@ -252,9 +254,8 @@ function textStyle(textsymbolizer, fature, type) {
  * @param {object} feature geojson feature, @see {@link http://geojson.org|geojson} Changed in version 0.0.4
  * @return ol.style.Style or array of it
  */
-export default function OlStyler(GeometryStyles, feature) {
-  const { properties, geometry } = feature;
-  const { type } = geometry;
+export default function OlStyler(GeometryStyles, type, feature) {
```

Comment:
I did not have time to review all changes yet, however about this one: the type argument was argument was removed in 0.0.4. It is contained in the feature object.

Reply:
Okay, you could get the type from the feature as well using

`const type = feature.getGeometry().getType ? feature.getGeometry().getType() : feature.geometry.type;`.

With the given code you support OpenLayers and plain geojson. Please tell me if this solution is fine for you.

I may want to ask, what is the background behind expecting a geojson feature in an _OlStyler_ function?

Reply:
We indeed plan to change the geojson requirement.

The feature argument could change into either geojson|olfeature like you did. That is fine.